### PR TITLE
Add ability to start RAP server in background.

### PR DESCRIPTION
**Pull request by m1ghtym0**
Solves issue #5338.
Used messages from HTTP background server.
Also adopted the one thread only convention.

### CODE REVIEW

File: libr/core/rtr.c [hunk 1]
```diff
@@ -828,6 +834,7 @@ R_API int r_core_rtr_http(RCore *core, int launch, const char *path) {
 	return ret;
 }
 
+
```
Comment (radare) [inline libr/core/rtr.c]: empty line

**Comment by m1ghtym0** [comment]
AppVeyor fails at smth. I haven't even touched, not sure what's happening there?

File: libr/core/rtr.c [hunk 2]
```diff
@@ -1188,15 +1196,41 @@ static void r_rap_packet_fill(ut8 *buf, const ut8* src, int len) {
 	}
 }
 
+static void r_core_rtr_rap_run(RCore *core, const char *input) {
+	r_core_cmdf (core, "o rap://%s", input);
+}
+
+static int r_core_rtr_rap_thread (RThread *th) {
+	if (!th) return false;
```
Comment (radare) [inline libr/core/rtr.c]: add braces on oneline conditionals too

File: libr/core/rtr.c [hunk 3]
```diff
@@ -1188,15 +1196,41 @@ static void r_rap_packet_fill(ut8 *buf, const ut8* src, int len) {
...
+	RapThread *rt = th->user;
+	if (!rt || !rt->core) return false;
```
Comment (radare) [inline libr/core/rtr.c]: same here, or just change the logic to nest the true conditional blocks

File: libr/core/rtr.c [hunk 4]
```diff
@@ -1188,15 +1196,41 @@ static void r_rap_packet_fill(ut8 *buf, const ut8* src, int len) {
...
+	}
+
+	if (*input=='&') {
+		if (rapthread) {
+			eprintf ("RAP Thread is already running\n");
+			eprintf ("This is experimental and probably buggy. Use at your own risk\n");
+		} else {
+			RapThread rt = { core, input + 1 };
+			rapthread = r_th_new (r_core_rtr_rap_thread, &rt, 0);
+			r_th_start (rapthread, 1);
```
Comment (radare) [inline libr/core/rtr.c]: maybe use true/false instead of 0,1 here, the signature of the function must be changed (I will do that later)

**Comment by Maijin** [comment]
For appveyor it's @radare fault no worries will be fixed soon ;)

**Comment by radare** [comment]
Just fix those tiny indent things and i'll merge. lgtm

**Comment by Maijin** [comment]
Tips: If you use Fix #XXXX in the commit message or title it will automatically close the corresponding issue

**Comment by m1ghtym0** [comment]
Thanks for reviews and tipps!

**Comment by radare** [comment]
Thanks!

**Pull request merged**